### PR TITLE
Use pytest to dump layers to geotiff files

### PR DESCRIPTION
**Pull request by kcartier-wri**
## Is this PR related to a JIRA ticket?
https://gfw.atlassian.net/browse/CIF-247

## What changes were proposed in this PR?
Extend testing

## How was this patch tested?
locally

### CODE REVIEW

**Comment by kcartier-wri** [comment]
Justin - As I mentioned, I'd like to include a way of dumping layers to geotiff files along with a QGIS project file for viewing them. What do you think of this current approach. I'm not fully satisfied with it, so am interested in your suggestions. Thanks.

File: tests/resources/layer_dumps_for_br_lauro_de_freitas/test_write_layers_to_geotiff.py
Comment (chrowe) [inline tests/resources/layer_dumps_for_br_lauro_de_freitas/test_write_layers_to_geotiff.py:1]: I get
```
FAILED tests/resources/layer_dumps_for_br_lauro_de_freitas/test_write_layers_to_geotiff.py::test_geotiff_writing - UnboundLocalError: local variable 'overture_buildings' referenced before assignment
```
When running 
`pytest tests/resources/layer_dumps_for_br_lauro_de_freitas/test_write_layers_to_geotiff.py`

**Review by chrowe — CHANGES_REQUESTED** [review]

File: tools/general_tools.py [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+import os
+
+def create_temp_folder(sub_directory):
+    if platform.system() == 'Linux':
+        path = '/tmp/%s' % sub_directory
+    elif platform.system() == 'Windows':
+        localappdata_path = os.getenv("LOCALAPPDATA")
+        path = r'%s\Temp\%s' % (localappdata_path, sub_directory)
+    else:
+        raise ValueError('Method not implemented for this OS.')
```
[the remark this answers is not in the document]
Comment (kcartier-wri) [reply tools/general_tools.py]: create_temp_folder function now uses tempfile library, so MacOS should now be handled. Please test or your machine.

File: city_metrix/layers/land_surface_temperature.py [hunk 2]
```diff
@@ -34,7 +34,8 @@ def apply_scale_factors(image):
         data = get_image_collection(ee.ImageCollection(l8_st), bbox, 30, "LST").ST_B10_mean
         return data
 
-    def write(self, output_path):
-        self.data.rio.to_raster(output_path)
+    # TODO: Is this inherited class function needed?
+    # def write(self, output_path):
+    #     self.data.rio.to_raster(output_path)
```
[the remark this answers is not in the document]
Comment (kcartier-wri) [reply city_metrix/layers/land_surface_temperature.py]: done

File: city_metrix/layers/high_land_surface_temperature.py [hunk 3]
```diff
@@ -55,7 +55,8 @@ def addDate(image):
         # convert to date object
         return datetime.datetime.strptime(hottest_date, "%Y%m%d").date()
 
-    def write(self, output_path):
-        self.data.rio.to_raster(output_path)
+    # TODO: Is this inherited class function needed?
+    # def write(self, output_path):
+    #     self.data.rio.to_raster(output_path)
```
[the remark this answers is not in the document]
Comment (kcartier-wri) [reply city_metrix/layers/high_land_surface_temperature.py]: done

File: tests/resources/layer_dumps_for_br_lauro_de_freitas/test_write_layers_to_geotiff.py
Comment (kcartier-wri) [reply tests/resources/layer_dumps_for_br_lauro_de_freitas/test_write_layers_to_geotiff.py:1]: I haven't been able to reproduce this issue and haven't even spotted the overture_buildings variable in the code. Also, the automatic execution in github passes without issue.

Could you isolate it further?

**Review by jterry64 — APPROVED** [review]
Looks good to me!

File: tests/resources/layer_dumps_for_br_lauro_de_freitas/test_write_layers_to_geotiff.py
Comment (chrowe) [reply tests/resources/layer_dumps_for_br_lauro_de_freitas/test_write_layers_to_geotiff.py:1]: I was able to run it with the latest changes

**Review by chrowe — APPROVED** [review]

File: city_metrix/layers/high_land_surface_temperature.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (kcartier-wri) [reply city_metrix/layers/high_land_surface_temperature.py]: done

**Pull request merged**